Security/JwtAuthentication.php

```diff
@@ -0,0 +1,166 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with DoctrineRestDriver.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace Circle\DoctrineRestDriver\Security;
+
+use Circle\DoctrineRestDriver\Types\Request;
+
+/**
+ * This file can be used in the driver's config to use
+ * the target API with basic http authentication
+ *
+ * @author    Liviu Oprisan
+ */
+class JwtAuthentication implements AuthStrategy {
+
+    /**
+     * @var array
+     */
+    private $config;
+
+    /**
+     * HttpBasicAuthentication constructor
+     *
+     * @param array $config
+     */
+    public function __construct(array $config) {
+        $this->config = $config;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function transformRequest(Request $request) {
+        
+        $jwtToken = $this->getJwtToken();
+        
+        $options  = $request->getCurlOptions();
+        
+        $concatenateBeforeToken = "";
+        if(isset($this->config['driverOptions']['jwt_prefix']))
+        {
+            $concatenateBeforeToken = $this->config['driverOptions']['jwt_prefix'];
+        }
+        
+        $headers  = empty($options[CURLOPT_HTTPHEADER]) ? [] : $options[CURLOPT_HTTPHEADER];
+        array_push($headers, 'X-API-Token: ' . $concatenateBeforeToken . $jwtToken);
+        $options[CURLOPT_HTTPHEADER] = $headers;
+
+        return $request->setCurlOptions($options);
+    }
+    
+    /**
+     * token is cached in session; it is revalidated "revalidate_token_time" seconds before expire 
+     * (defaults to 5 seconds)
+     * @return string
+     */
+    private function getJwtToken()
+    {
+        if(!isset($_SESSION["doctrine_rest_driver_jwt_token"]))
+        {
+            $_SESSION["doctrine_rest_driver_jwt_token"] = $this->getJwtTokenFromRemote();
+        }
+        else if($this->shouldRefreshToken($_SESSION["doctrine_rest_driver_jwt_token"]))
+        {
+            $_SESSION["doctrine_rest_driver_jwt_token"] = $this->getJwtTokenFromRemote();
+        }
+        
+        return $_SESSION["doctrine_rest_driver_jwt_token"];
+    }
+    
+    /**
+     * checks if token is about to expire - 5 seconds before expire date 
+     * or "revalidate_token_time" before expire from config -> dbal ->
+     * -> connections -> options -> revalidate_token_time, if set
+     * @param string $token
+     * @return boolean
+     */
+    private function shouldRefreshToken($token)
+    {
+        
+        $elements = explode('.', $token);
+        $payload = json_decode(base64_decode($elements[1]));
+        
+        $expireTimestamp = $payload->iat;
+        $revalidateTime = 5;
+        if(isset($this->config['driverOptions']['revalidate_token_time']))
```

Comment:
$revalidateTime = isset($this->config['driverOptions']['revalidate_token_time']) ? $this->config['driverOptions']['revalidate_token_time'] : 5;